tests/acceptance/apps_dns/apps_dns_test.go

```diff
@@ -0,0 +1,309 @@
+			ips := []net.IP{
+				net.ParseIP("10.11.12.13"),
+				net.ParseIP("10.11.12.14"),
+			}
+			expectedAnswer := []dns.RR{
+				&dns.A{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 4,
+					},
+					A: ips[0].To4(),
+				},
+				&dns.A{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 4,
+					},
+					A: ips[1].To4(),
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type A")
+			res, err := dnsQuery(domainName, dns.TypeA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(sortedCopy(res.Answer)).To(Equal(sortedCopy(expectedAnswer)))
+		})
+
+		It("should resolve to multiple IPv6s", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{
+				net.ParseIP("2001:db8::68"),
+				net.ParseIP("2001:db8::69"),
+			}
+			expectedAnswer := []dns.RR{
+				&dns.AAAA{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeAAAA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 16,
+					},
+					AAAA: ips[0],
+				},
+				&dns.AAAA{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeAAAA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 16,
+					},
+					AAAA: ips[1],
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type AAAA")
+			res, err := dnsQuery(domainName, dns.TypeAAAA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(sortedCopy(res.Answer)).To(Equal(sortedCopy(expectedAnswer)))
+		})
+
+		It("should resolve when the request is of type AAAA and the discovered service has an IPv4 address", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{net.ParseIP("10.11.12.13")}
+			expectedAnswer := []dns.RR{
+				&dns.AAAA{
+					Hdr: dns.RR_Header{
+						Name:     domainName,
+						Rrtype:   dns.TypeAAAA,
+						Class:    dns.ClassINET,
+						Ttl:      ttl,
+						Rdlength: 16,
+					},
+					AAAA: ips[0].To16(),
+				},
+			}
+			sdc.Handle(domainName, fake.Handler(ips))
+
+			// Assert
+			By("sending a question type AAAA")
+			res, err := dnsQuery(domainName, dns.TypeAAAA)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res.Rcode).To(Equal(dns.RcodeSuccess))
+			Expect(res.Answer).To(Equal(expectedAnswer))
+		})
+
+		It("should resolve when the request is of type A and the discovered service has an IPv6 that's a valid IPv4", func() {
+			// Prepare
+			domainName := dns.Fqdn(fmt.Sprintf("%d.apps.internal", rand.Uint64()))
+			ips := []net.IP{net.ParseIP("0:0:0:0:0:ffff:a0b:c0d")}
```

Comment:
```suggestion
			ips := []net.IP{net.ParseIP("::ffff:a0b:c0d")}
```
If I understand my IPv6 shorthands correctly.